codegen/testserver/generated_test.go

```diff
@@ -276,67 +444,84 @@ func TestGeneratedServer(t *testing.T) {
 		}, tracerLog)
 	})
 
-	t.Run("subscriptions", func(t *testing.T) {
-		t.Run("wont leak goroutines", func(t *testing.T) {
-			initialGoroutineCount := runtime.NumGoroutine()
-
-			sub := c.Websocket(`subscription { updated }`)
+	t.Run("take ctx over from prev step", func(t *testing.T) {
+		resolvers := &testResolver{tick: make(chan string, 1)}
 
-			resolvers.tick <- "message"
+		steps := []string{
+			"StartOperationParsing",
+			"StartOperationValidation",
+			"StartOperationExecution",
+			"StartFieldExecution",
+			"StartFieldResolverExecution",
+			"StartFieldChildExecution",
+		}
 
-			var msg struct {
-				resp struct {
-					Updated string
+		assertStep := func(target string) func(ctx context.Context) {
+			return func(ctx context.Context) {
+				for _, step := range steps {
+					assert.NotEmpty(t, ctx.Value(step))
+					if step == target {
```

Comment:
I still think its better to write this out the long way in the test, so it is easy to understand what you're testing. It took me a few reads to get it.